packages/block-library/src/columns/edit.js

```diff
@@ -102,17 +102,29 @@ function ColumnsEditContainer( {
 			</InspectorControls>
 			{ InspectorControlsColorPanel }
 			<BackgroundColor>
-				<TextColor>
-					<InnerBlocks
-						allowedBlocks={ ALLOWED_BLOCKS }
-						__experimentalMoverDirection="horizontal"
-						ref={ ref }
-						__experimentalTagName={ Block.div }
-						__experimentalPassedProps={ {
-							className: classes,
-						} }
-					/>
-				</TextColor>
+				{ ( backgroundProps ) => (
+					<TextColor>
```

Comment:
@jorgefilipecosta Do you think you can help to update the useColors API to avoid producing multiple components and just return a single `ApplyColors` instead? I think that flexibility is harming the API and making things more complex than they should be.

Reply:
I don't love the implicitness of these components. It's very unclear how these work at first.

Reply:
I don't either, though merging style and classname props can be tricky if left to the consumer. I believe that's the main reason for the implicitness. 

I do believe though that we should have a single component and not multiple ones.

Reply:
Implicitness can be good if we have multiple hooks:

```
ApplyColors
ApplyFontSizes
ApplyGlobalStyles
```

merging could become a nightmare.

Reply:
> @jorgefilipecosta Do you think you can help to update the useColors API to avoid producing multiple components and just return a single ApplyColors instead? I think that flexibility is harming the API and making things more complex than they should be.


We can plan on making this change but I think it is a big API change. Currently "TextColor" is a component that applies styles and classes to descendants but TextColor.setColor contains a function to change the TextColor value, TextColor.slug displays the slug of the color. What would be the ideal return API? Should we return ApplyStyles and then have ApplyStyles.textColor.setColor? Should we return an object with { ApplyStyles, textColor, backgroundColor } ?
The colors may not all be applied to the same elements, in this case, the expectation is that users will use multiple hook calls?
Doing this with ApplyColors also some use cases more difficult. For example, a case where we have a backgroundColor applied to a div, and textColor applied to a descendent of the div will require two multiple hook calls and in that case, contrast checking will not work right away (the developer will need to use the contrast checking components).

Reply:
`{ ApplyStyles, textColor, backgroundColor }` This is closer to what I think the AAPI should be but before returning these extra values, It would be good to understand if and how they're used today.

> The colors may not all be applied to the same elements, in this case, the expectation is that users will use multiple hook calls?

This is very rare but yes.

> Doing this with ApplyColors also some use cases more difficult. For example, a case where we have a backgroundColor applied to a div, and textColor applied to a descendent of the div will require two multiple hook calls and in that case, contrast checking will not work right away (the developer will need to use the contrast checking components).

Contrast checking could be its own hook that is used internally by useColors or used externaly;

--- 
I agree that both approaches have drawbacks but we should monitor the existing use-cases, I believe it will become clearer which API is best.